### PR TITLE
Add new ContactExpandableGroup block to LearnPages

### PR DESCRIPTION
**Pull request by chosak**
This change introduces a new StreamField block named ContactExpandableGroup, and adds it as an option on LearnPages. See GHE/el-camino#21 for background on motivations for this change; this will make it easier to support [the site Contact Us page](https://www.consumerfinance.gov/about-us/contact-us/). Ping @sonnakim.

This branch has been deployed to our internal DEV3 testing server, and can be viewed in action at DEV3/about-us/contact-us/.

## Additions

- New `ContactExpandableGroup` organism that renders similarly to regular `ExpandableGroup`s, except that the content gets auto-filled from `Contact` snippets instead of from manual user input.

## Changes

- A new option has been added to `ExpandableGroup`, which wraps the entire organism in a `<div>` with an `id` based on the slugified header. This allows for in-page linking to a specific `ExpandableGroup`. I've left this disabled by default. Question for @anselmbradford @Scotchester: any issues with adding the plain `<div>` to all EGs even if this is disabled?
- I've modified the `Contact` snippet so that it always sorts by `heading` by default, which seems reasonable and prevents us from needing to manually specify it each time.

## Testing

1. See new unit tests, run `tox`.
2. Try creating a LearnPage and adding a `ContactExpandableGroup` block.

## Screenshots

New "Contact expandable group" block available on LearnPages:

image

Wagtail editing interface:

image

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: cfgov/v1/atomic_elements/organisms.py [hunk 1]
```diff
@@ -802,6 +804,62 @@ class Media:
...
+        return context
+
+    def bulk_to_python(self, values):
+        """Support bulk retrieval of Contacts to reduce database queries.
+
+        This method leverages Wagtail's undocumented method for doing bulk
+        retrieval of data for StreamField blocks. It retrieves all Contacts
+        referenced by a StreamField in a single lookup, instead of the
+        default behavior of doing one database query per SnippetChooserBlock.
+        """
```
Comment (higs4281) [inline cfgov/v1/atomic_elements/organisms.py]: A useful tactic to be aware of.

File: cfgov/v1/tests/atomic_elements/organisms/test_contact_expandable_group.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            'contacts': [1, 2],
+        })
+        html = block.render(value)
+        self.assertTagInHTML(html, '<div id="wyld-stallyns">')
+        self.assertIn('Bill', html)
+        self.assertIn('Ted', html)
+
+    def test_bulk_to_python(self):
+        Contact.objects.bulk_create(
+            Contact(pk=i, heading=str(i)) for i in range(10)
```
Comment (higs4281) [inline cfgov/v1/tests/atomic_elements/organisms/test_contact_expandable_group.py]: I had always assumed you couldn't have a primary key of zero, but not so!

File: cfgov/v1/tests/atomic_elements/organisms/test_contact_expandable_group.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                        'contacts': list(range(5, 10))
+                    },
+                },
+            ],
+            is_lazy=True
+        )
+
+        # This verifies that the block does a single database query to retrieve
+        # all 10 Contacts, even though they are split up between two different
+        # ContactExpandableGroup blocks.
```
Comment (higs4281) [inline cfgov/v1/tests/atomic_elements/organisms/test_contact_expandable_group.py]: Nice!

**Comment by Scotchester** [comment]
> A new option has been added to `ExpandableGroup`, which wraps the entire organism in a `<div>` with an `id` based on the slugified header. This allows for in-page linking to a specific `ExpandableGroup`.

I would've just added it to the `<h3>` tag, rather than introducing a new wrapping `<div>`, but there's nothing inherently wrong with the approach you took.

> I've left this disabled by default. Question for @anselmbradford @Scotchester: any issues with adding the plain `<div>` to all EGs even if this is disabled?

No concern with doing that. I would go even further and fulfill my dream of having middleware that adds a slugified `id`s to all rendered heading elements on the site 😀

**Comment by anselmbradford** [comment]
It would be nice to move this into wagtail groups for expandables, since it doesn't begin with expandable https://github.com/cfpb/cfgov-refresh/pull/5236

**Pull request merged**